keras_nlp/utils/text_generation.py

```diff
@@ -160,6 +160,154 @@ def token_probability_fn(inputs):
     return prompt
 
 
+def beam_search(
+    token_probability_fn,
+    prompt,
+    max_length,
+    beam_width,
+    from_logits=False,
+    end_token_id=None,
+    pad_token_id=0,
+):
+    """
+    Text generation utility based on beam search algorithm.
+
+    At each time-step, beam search keeps the top `beam_width` beams (sequences),
+    and uses each one of the beams to predict candidate next tokens. The top
+    `beam_width` most probable next tokens are kept and appended to their
+    respective beams, before beginning the next iteration.
+
+    Args:
+        token_probability_fn: a callable, which takes in input_sequence
```

Comment:
We should document the input shape and output shape expected by this function, as it is a non-standard batch size.